source/appModules/code.py

```diff
@@ -21,11 +24,95 @@ class VSCodeDocument(Document):
 	_get_treeInterceptorClass = NVDAObject._get_treeInterceptorClass
 
 
+DIGIT_EXPR = re.compile(r"\d+")
+
+
 class AppModule(appModuleHandler.AppModule):
+	def __init__(self, *args, **kwargs):
+		super().__init__(*args, **kwargs)
+		self._status = None
+
+	@staticmethod
+	def _search_for_statusbar(root: NVDAObject) -> NVDAObject | None:
+		seen = set()
+		t = deque((root,))
+		while t:
+			obj = t.popleft()
+			if obj in seen:
+				continue
+			seen.add(obj)
+			try:
+				if obj.role == controlTypes.Role.STATUSBAR:
+					return obj
+				# IA2 ID often contains "statusbar"
+				ia2id = obj.IA2Attributes.get("id")
+				if ia2id and "statusbar" in ia2id.casefold():
+					return obj
```

Comment:
What do we expect to raise an Exception? `obj.children`?
What kind of Exception do we expect to be raised?
Can the rest of this be moved out of the try/except?

Reply:
Yeah, it'd probably be better to log these.